app/controllers/items_controller.rb

```diff
@@ -19,6 +19,16 @@ def update
     render :edit
   end
 
+  def reserve
+    item = Item.find(params[:item_id])
+    wishlist = Wishlist.find(params[:wishlist_id])
+    authorize! :reserve, item
+    Item.change_reserved(item, params[:item][:is_reserved])
+    redirect_to wishlists_for_guest_path(wishlist.token), notice: t(".success")
+  rescue CanCan::AccessDenied
+    redirect_to wishlists_for_guest_path(wishlist.token), alert: t(".error")
```

Comment:
Meh. This won't be thrown. Everyone can reserve, so why even bother catching this?